gen_errs.py

```diff
@@ -0,0 +1,26 @@
+# Updates error codes in ccp_error.h
```

Comment:
maybe use `__COUNTER__` instead? You do this grouping though, so I guess it is fine as is.

Reply:
https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html

Reply:
ah i wasn't aware of this. the main reason for doing it manually was so that, given a return error code, you could easily look up where it came from. if we were actively adding lots of these i agree counter would be better, but i doubt we'll make many changes to this frequently